pajbot/modules/basic/roll.py

```diff
@@ -0,0 +1,158 @@
+        ModuleSetting(
+            key="timeout_unit",
+            label="Choose the timeout format to use. Maximum Twitch timeout limits are enforced.",
+            type="options",
+            required=False,
+            default="Minutes",
+            options=["Seconds", "Minutes", "Hours", "Days", "Weeks"],
+        ),
+        ModuleSetting(
+            key="zero_response",
+            label="Additional text to say when the user rolls a 0. Timeouts must be enabled.",
+            type="text",
+            required=False,
+            placeholder="You're safe! For now... PRChase",
+            default="You're safe! For now... PRChase",
+            constraints={"max_str_len": 100},
+        ),
+    ]
+
+    def load_commands(self, **options):
+        self.commands["roll"] = Command.raw_command(
+            self.roll,
+            sub_only=self.settings["subscribers_only"],
+            delay_all=self.settings["global_cd"],
+            delay_user=self.settings["user_cd"],
+            level=self.settings["level"],
+            command="roll",
+            examples=[
+                CommandExample(
+                    None,
+                    "Roll a random number",
+                    chat="user:!roll\n" "bot:@pajlada You rolled a 9!",
+                    description="",
+                ).parse(),
+            ],
+        )
+
+    # We're converting timeout times to seconds in order to avoid having to specify the unit to Twitch
+    def seconds_conversion(self, rolled_value):
+        if rolled_value != 0:
+            if self.settings["timeout_unit"] == "Seconds":
+                return rolled_value
+            elif self.settings["timeout_unit"] == "Minutes":
+                return rolled_value * 60
+            elif self.settings["timeout_unit"] == "Hours":
+                return rolled_value * 3600
+            elif self.settings["timeout_unit"] == "Days":
+                return rolled_value * 86400
+            elif self.settings["timeout_unit"] == "Weeks":
+                return rolled_value * 604800
+        else:
+            return True
+
+    def roll(self, event, source):
+        if self.settings["subscribers_only"] and not source.subscriber:
+            return True
+
+        rolled_value = random.randint(self.settings["low_value"], self.settings["high_value"])
+
+        if self.settings["enable_timeouts"] is True:
+            if rolled_value == 0 and self.settings["zero_response"] != "":
+                self.bot.send_message_to_user(
+                    source, f"You rolled a {rolled_value}! {self.settings['zero_response']}", event, method="reply"
+                )
+            else:
+                timeout_length = self.seconds_conversion(rolled_value)
+                # Check if timeout value is over Twitch's maximum
+                if timeout_length > 1209600:
+                    timeout_length = 1209600
+
+                self.bot.timeout(source, timeout_length, f"You rolled a {rolled_value}!")
```

Comment:
check for mod/broadcaster

Reply:
I was under the impression that pajbot automatically checked for this in the timeout function, but I can't seem to find it anywhere. Will have to talk to pajlada about why that is.